### PR TITLE
Update warning, add tests for project requirements check

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

* Make warning more general for differences between PEP 508 and pip
* Add tests for _check_requirements

### Types of change

?

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/tests/test_cli.py [hunk 1]
```diff
@@ -855,3 +857,35 @@ def test_span_length_freq_dist_output_must_be_correct():
...
+    assert output == (False, False)
+
+    # excessive guard against unlikely package name
+    try:
+        pkg_resources.require("spacyunknowndoesnotexist12345")
+    except pkg_resources.DistributionNotFound:
+        reqs = """# comment
+        spacyunknowndoesnotexist12345"""
+        output = _check_requirements([req.strip() for req in reqs.split("\n")])
+        assert output == (True, False)
```
Comment (rmitsch) [inline spacy/tests/test_cli.py:891]: The point of the `try`/`except` is to verify first that the unlikely package is not installed, right? Alternative suggestion, feel free to ignore:
```suggestion
    pkg_name = "spacyunknowndoesnotexist12345"
    with pytest.raises(pkg_resources.DistributionNotFound):
        pkg_resources.require(pkg_name)
    reqs = f"# comment\n{pkg_name}"
    output = _check_requirements([req.strip() for req in reqs.split("\n")])
    assert output == (True, False)
```

**Review by rmitsch — APPROVED** [review]

File: spacy/tests/test_cli.py (hunk 1 above)
Comment (adrianeboyd) [reply spacy/tests/test_cli.py:891]: I might be confused about what you're trying to propose, but the point of this was to completely skip the test in the (rare, unlikely) case that someone intentionally created an venv with this package in order to break the test suite.

Comment (rmitsch) [reply spacy/tests/test_cli.py:891]: Ah, I thought we wanted to guarantee that this package is not installed. Skipping makes more sense.

**Comment by adrianeboyd** [comment]
The test is now skipped entirely if you have this unlikely package installed, but I think I can live with this.

**Pull request merged**